pkg/backends/newrelic/newrelic.go

```diff
@@ -295,6 +295,26 @@ func (n *Client) Name() string {
 	return BackendName
 }
 
+// RetryAfterError indicates if we should wait before retrying
+type RetryAfterError struct {
+	Response *http.Response
```

Comment:
This mixes concerns - the error is about "retry the operation after a provided interval", but the implementation is "here's an http response that failed, figure out the rest yourself".

The parsing of the header should be done immediately, and `RetryAfterError` should just contain a `time.Duration`.  It also simplifies the error handling, because if you can't parse it, or the header is missing, you can fall back to the default error path.

Reply:
I've addressed this in last commit.